test/e2e/vsphere_volume_diskformat.go

```diff
@@ -0,0 +1,211 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"os"
+	"path/filepath"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/vim25/types"
+	"golang.org/x/net/context"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8stype "k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
+	vsphere "k8s.io/kubernetes/pkg/cloudprovider/providers/vsphere"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+/*
+	Test to verify diskformat specified in storage-class is being honored while volume creation.
+	Valid and supported options are eagerzeroedthick, zeroedthick and thin
+
+	Steps
+	1. Create StorageClass with diskformat set to valid type
+	2. Create PVC which uses the StorageClass created in step 1.
+	3. Wait for PV to be provisioned.
+	4. Wait for PVC's status to become Bound
+	5. Create pod using PVC on specific node.
+	6. Wait for Disk to be attached to the node.
+	7. Get node VM's devices and find PV's Volume Disk.
+	8. Get Backing Info of the Volume Disk and obtain EagerlyScrub and ThinProvisioned
+	9. Based on the value of EagerlyScrub and ThinProvisioned, verify diskformat is correct.
+	10. Delete pod and Wait for Volume Disk to be detached from the Node.
+	11. Delete PVC, PV and Storage Class
+*/
+
+var _ = framework.KubeDescribe("Volume Disk Format [Feature:Volume]", func() {
```

Comment:
There's been some back and forth here. IMO, this line should be:
`var _ = framework.KubeDescribe("Volume Disk Format [Volumes]", func() {...`
In a previous comment I've shown about 25x more occurences in _test/_ of my suggestion vs what you have above. But I may be misunderstanding how [Feature:xyz]` works, or vsphere may have special needs here.